src/test/java/buisness/core/api/FiltersRestClient.java

```diff
@@ -0,0 +1,101 @@
+package buisness.core.api;
+
+import core.dto.api.FilterDto;
+import core.enums.TestCacheKey;
+import core.utils.TestCache;
+import io.restassured.response.ValidatableResponse;
+import lombok.extern.slf4j.Slf4j;
+
+import static core.enums.PropertyNames.BASE_URL;
+import static core.utils.services.ConfigurationService.getProperty;
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+import static java.util.Objects.requireNonNull;
+
+@Slf4j
+public class FiltersRestClient extends ApiRestClient {
+
+  private static final String FILTERS_PATH = "api/v1/%s/filter";
+
+  private static String getUrlForUpdate() {
```

Comment:
The method is not needed in case if you use all features of Rest Assured. For example, base URL you can path to the baseUri method and the path to the 'basePath' method like a placeholder 'api/v1/{projectId}/filter' with the chain method pathParam("projectId", projectId) and in method get() path just filterId
Example:

  private static final String FILTERS_PATH = "api/v1/{projectId}/filter";
  
  RequestSpecification rs = given()
    .when().baseUri(getProperty(BASE_URL.getName()))
    .accept(JSON)
    .contentType(JSON)
    .auth().oauth2(getBearerToken());

  public void getAllFilters() {
    String project = TestCache.get(TestCacheKey.PROJECT, String.class);
    ValidatableResponse response = given().spec(rs)
      .basePath(FILTERS_PATH).pathParam("projectId", project)
      .get()
      .then().log().all();

    TestCache.putDataInCache(TestCacheKey.RESPONSE, response);
  }

Reply:
yes, replaced to RequestSpecification